h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java

```diff
@@ -438,13 +442,94 @@ private void scoreAndBuildTrees(final XGBoostSetupTask setupTask, final BoosterP
           break;
         }
       }
+
+      Map<String, Integer> monotoneConstraints = _parms.monotoneConstraints();
+      if ((!monotoneConstraints.isEmpty()) && constraintCheckEnabled()) {
+        checkConstraints(model.model_info(), monotoneConstraints);
```

Comment:
Don't you need to use the `boosterProvider` to get the booster in this phase of training?

Reply:
true, added call to get the latest booster